packages/config/tests/validate/tests.js

```diff
@@ -12,10 +12,6 @@ test('plugins: not array of objects', async (t) => {
   await runFixture(t, 'plugins_not_objects')
 })
 
-test('plugins: do not allow duplicates', async (t) => {
```

Comment:
Should we still forbid duplicates and only allow duplicates when overriding a plugin via `context`?

```toml
[[plugins]]
package = "./plugin.js"

[[plugins]]
package = "./plugin.js"
```

Does't seem like a valid configuration

Reply:
I was wondering about that one too.

We will be allowing the following to be merged: 
  - Different origins (plugin defined in both UI and `netlify.toml`)
  - Different contexts (no context and current context)

I was wondering if there could some reasons why users would want to incrementally specify their plugins in two different blocks in their `netlify.toml`? For example, TOML does not allow importing a shared configuration. One way to solve this could be to prepend a shared-projects `netlify.toml` to a project-specific `netlify.toml` at build time. 

On the other hand, I could also see the above pattern being an indication the user forgot that they already added/configured a plugin on top on their `netlify.toml`.

What are your thoughts on this?

Reply:
I remember us supporting this syntax, but for another use case - of running the same plugin multiple times.

> One way to solve this could be to prepend a shared-projects `netlify.toml` to a project-specific `netlify.toml` at build time.

I think using a place holder and a find replace or even parsing the `toml` file and serializing it again after updating a specific plugin configuration should work too without the possible confusion created by appending.

> On the other hand, I could also see the above pattern being an indication the user forgot that they already added/configured a plugin on top on their `netlify.toml`.

💯  I can see this happening

Reply:
Good points. I have added the validation back.

This might make adding context-specific plugins configuration a little harder to implement because both `config.plugins` and `config.context.{context}.plugins` are defined in `netlify.toml`, and they would be using the same merging code. But there is probably a way around it. :+1:

Reply:
Done at https://github.com/netlify/build/pull/2566